### ISSUE TITLE
Feature: announce Upgrade possibilities

### ISSUE THREAD

**Issue by icing** [icing/mod_h2#54]
Announce other supported protocols when a client talks http/1.1, but may `Upgrade:`. Since `Upgrade:` is a connection header, clients need to see this only on the first request on a connection.

The purpose of this feature is to make it easy for clients to detect server capabilities without wasting resources on every request. It also might help client's connection management where http/1.1 clients will first open several connection (comonly 6) and then later might want to fall back to a single one using http/2. Upgrading all 6 connections to h2c might be undesirable.

**Comment by icing** [comment]
Added in trunk - waiting for the outcry.

**Comment by icing** [comment]
in /trunk: announce protocols to possibly upgrade to
* on HTTP/1.1 connections only
* announce only protocols that have higher preference
* announce only on first request of connection

If one makes a HTTP/1.1 request to a site with
```
Protocols h2c http/1.1
```
then the first response will include
```
Upgrade: h2c
Connection: Upgrade
```
among its headers.

A HTTP/1.1 request to a site configured with
```
Protocols http/1.1 h2c
```
will not show an ```Upgrade```header as the connection is already on the most preferred protocol.

**Comment by icing** [comment]
Available for testing in v1.0.3

**Comment by icing** [comment]
released in 2.4.18
